src/@apollo/client/cache/inmemory/ApolloClient__Cache_InMemory_Policies_FieldPolicy.re

```diff
@@ -7,12 +11,77 @@ module FieldMergeFunction = {
   type t = Js_.t;
 };
 
+module StorageType = {
+  type t = Js.Dict.t(Js.Json.t);
+  module Js_ = {
+    type nonrec t = t;
+  };
+};
+
+module FieldFunctionOptions = {
+  type useMethodFunctionInThisModuleInstead;
+  module Js_ = {
+    // export interface FieldFunctionOptions<TArgs = Record<string, any>, TVars = Record<string, any>> {
+    //     args: TArgs | null;
+    //     fieldName: string;
+    //     storeFieldName: string;
+    //     field: FieldNode | null;
+    //     variables?: TVars;
+    //     isReference: typeof isReference;
+    //     toReference: ToReferenceFunction;
+    //     storage: StorageType | null;
+    //     cache: InMemoryCache;
+    //     readField: ReadFieldFunction;
+    //     canRead: CanReadFunction;
+    //     mergeObjects<T extends StoreObject | Reference>(existing: T, incoming: T): T | undefined;
+    // }
+    type t = {
+      args: Js.nullable(Js.Dict.t(Js.Json.t)),
+      fieldName: string,
+      storeFieldName: string,
+      field: Js.nullable(FieldNode.t),
+      variables: option(Js.Dict.t(Js.Json.t)),
+      isReference: bool,
+      toReference: useMethodFunctionInThisModuleInstead,
+      storage: Js.nullable(StorageType.Js_.t),
+      cache: ApolloCache.Js_.t(Js.Json.t),
+      readField: useMethodFunctionInThisModuleInstead,
+      canRead: useMethodFunctionInThisModuleInstead,
+      mergeObjects: useMethodFunctionInThisModuleInstead,
+    };
+  };
+
+  type t =
+    Js_.t = {
+      args: Js.nullable(Js.Dict.t(Js.Json.t)),
+      fieldName: string,
+      storeFieldName: string,
+      field: Js.nullable(FieldNode.t),
+      variables: option(Js.Dict.t(Js.Json.t)),
+      isReference: bool,
+      toReference: useMethodFunctionInThisModuleInstead,
+      storage: Js.nullable(StorageType.t),
+      cache: ApolloCache.Js_.t(Js.Json.t),
+      readField: useMethodFunctionInThisModuleInstead,
+      canRead: useMethodFunctionInThisModuleInstead,
+      mergeObjects: useMethodFunctionInThisModuleInstead,
+    };
+
+  [@bs.send] external canRead: t => CanReadFunction.Js_.t = "canRead";
+  [@bs.send]
+  external mergeObjects:
+    (t, ~existing: Js.Json.t, ~incoming: Js.Json.t) => option(Js.Json.t) =
+    "canRead";
+  [@bs.send] external readField: t => ReadFieldFunction.Js_.t = "readField";
+  [@bs.send] external toReference: t => ToReferenceFunction.t = "toReference";
+};
+
 module FieldReadFunction = {
-  type t;
+  type t = (option(Js.Json.t), FieldFunctionOptions.t) => Js.Json.t;
```

Comment:
🤔 Perhaps Json is not the right type here. If I remember correctly you could use functions with `makeVar`.